fn_utilities/fn_utilities/components/utilities_attachment_to_base64.py

```diff
@@ -23,35 +23,53 @@ def _attachment_to_base64_function(self, event, *args, **kwargs):
             incident_id = kwargs.get("incident_id")  # number
             task_id = kwargs.get("task_id")  # number
             attachment_id = kwargs.get("attachment_id")  # number
+            artifact_id = kwargs.get("artifact_id") # number
 
             log.info("incident_id: %s", incident_id)
             log.info("task_id: %s", task_id)
             log.info("attachment_id: %s", attachment_id)
+            log.info("artifact_id: %s", artifact_id)
+            
             if incident_id is None and task_id is None:
                 raise FunctionError("Error: incident_id or task_id must be specified.")
-            if attachment_id is None:
+            if artifact_id is None and attachment_id is None:
```

Comment:
I don't believe this is correct to use 'and' logic'. An attachment can be one of three types:
- incident level attachment
- artifact (of type other file)
- task level attachment

so artifact_id and attachment_id are mutually exclusive. task_id and artifact_id are mutually exclusive

Reply:
I understand the confusion, but had to choose this logic since attachment type artifact(other file and log file) has no attribute called attachment in pre-processing script.